google/resource_kms_key_ring_test.go

```diff
@@ -0,0 +1,78 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"log"
+)
+
+func TestAccGoogleKmsKeyRing_basic(t *testing.T) {
+	name := acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testGoogleKmsKeyRing_recreate(name),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testGoogleKmsKeyRing_basic(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleKmsKeyRingExists("google_kms_key_ring.key_ring"),
+				),
+			},
+		},
+	})
+}
+
+func testGoogleKmsKeyRing_basic(name string) string {
```

Comment:
I don't have a good reason behind this, but we tend to keep these functions at the end of the file